### DISCUSSION TITLE
hyper-express框架集成node-easywechat问题

### DISCUSSION THREAD

**Discussion by uctoo** [hpyer/node-easywechat#94] [General]
没能搞定，作者能看看怎么解决吗？https://gitee.com/UCT/uctoo-backend/blob/master/src/app/controllers/wechatopen_applet/index.ts

**Comment by hpyer** [comment]
不好意思，这两天有点忙，才看到。你具体遇到的问题是什么？

**Reply by uctoo** [reply]
hyper-express 是基于uWebSockets.js的，和http.js的IncomingMessage实现有区别。
初始化时
let request = new ServerRequest.createFromIncomingMessage(req);
报这个错误
Argument of type 'Request<DefaultRequestLocals>' is not assignable to parameter of type 'IncomingMessage'.
  Type 'Request<DefaultRequestLocals>' is missing the following properties from type 'IncomingMessage': aborted, httpVersion, httpVersionMajor, httpVersionMinor, and 9 more.ts(2345)

**Reply by hpyer** [reply]
好的，晚点我写个脚本跑下。。大概率是要另外封装个方法兼容这个问题了。

**Comment by hpyer** [comment]
试了一下，不要用 `ServerRequest.createFromIncomingMessage(req)` 这个静态方法去创建 request 对象，改用 `new ServerRequest()` 的方式手工创建就可以的。代码如下：

```js
const { OfficialAccount, ServerRequest } = require('node-easywechat');
const HyperExpress = require('hyper-express');

const app = new OfficialAccount({
  app_id: '',
  secret: '',
  token: '',
  aes_key: '',
});

const webserver = new HyperExpress.Server();

webserver.any('/server', async (req, resp) => {

  const body = await req.text();

  const request = new ServerRequest(
    req.method,
    req.url,
    req.headers,
    body,
    '1.1',
    req.query_parameters
  );
  app.setRequest(request);

  const server = app.getServer();

  server.with(async (message) => {
    console.log('message', message);
  });

  const response = await server.serve();

  resp.send(response.getBody());
})

// Activate webserver by calling .listen(port, callback);
webserver.listen(6000)
  .then((socket) => console.log('Webserver started on port 6000'))
  .catch((error) => console.log('Failed to start webserver on port 6000'));

```

**Reply by uctoo** [reply]
thank you，我记得我用构造函数试过，那我再试试
